app/src/androidTest/java/com/android/unio/end2end/ResetPasswordSettingsTest.kt

```diff
@@ -0,0 +1,86 @@
+package com.android.unio.end2end
+
+import android.util.Log
+import androidx.compose.ui.test.isDisplayed
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.test.filters.LargeTest
+import com.android.unio.model.preferences.AppPreferences
+import com.android.unio.model.strings.test_tags.BottomNavBarTestTags
+import com.android.unio.model.strings.test_tags.HomeTestTags
+import com.android.unio.model.strings.test_tags.SettingsTestTags
+import com.android.unio.model.strings.test_tags.UserProfileTestTags
+import dagger.hilt.android.testing.HiltAndroidTest
+import java.net.URL
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.json.JSONObject
+import org.junit.Test
+
+@LargeTest
+@HiltAndroidTest
+class ResetPasswordSettingsTest : EndToEndTest() {
+
+  @Test
+  fun testUserCanResetPassword() {
+    signInWithUser(composeTestRule, MarjolaineLemm.EMAIL, MarjolaineLemm.OLD_PASSWORD)
+
+    composeTestRule.waitUntil(10000) {
+      composeTestRule.onNodeWithTag(HomeTestTags.SCREEN).isDisplayed()
+    }
+
+    composeTestRule.onNodeWithTag(BottomNavBarTestTags.MY_PROFILE).performClick()
+    composeTestRule.waitUntil(1000) {
+      composeTestRule.onNodeWithTag(UserProfileTestTags.SCREEN).isDisplayed()
+    }
+
+    composeTestRule.onNodeWithTag(UserProfileTestTags.SETTINGS).performClick()
+    composeTestRule.waitUntil(1000) {
+      composeTestRule.onNodeWithTag(UserProfileTestTags.BOTTOM_SHEET).isDisplayed()
+    }
+    composeTestRule.onNodeWithTag(UserProfileTestTags.USER_SETTINGS).performClick()
+    composeTestRule.waitUntil(1000) {
+      composeTestRule.onNodeWithTag(SettingsTestTags.SCREEN).isDisplayed()
+    }
+
+    composeTestRule.onNodeWithTag(AppPreferences.RESET_PASSWORD).performClick()
+
+    Thread.sleep(1000)
+
+    simulateResetPassword()
+
+    composeTestRule.onNodeWithTag(SettingsTestTags.GO_BACK).performClick()
+    composeTestRule.waitUntil(1000) {
+      composeTestRule.onNodeWithTag(UserProfileTestTags.SCREEN).isDisplayed()
+    }
+    composeTestRule.onNodeWithTag(UserProfileTestTags.SETTINGS).performClick()
+    composeTestRule.waitUntil(1000) {
+      composeTestRule.onNodeWithTag(UserProfileTestTags.BOTTOM_SHEET).isDisplayed()
+    }
+    composeTestRule.onNodeWithTag(UserProfileTestTags.SIGN_OUT).performClick()
+
+    signInWithUser(composeTestRule, MarjolaineLemm.EMAIL, MarjolaineLemm.NEW_PASSWORD)
+
+    composeTestRule.waitUntil(10000) {
+      composeTestRule.onNodeWithTag(HomeTestTags.SCREEN).isDisplayed()
+    }
+
+    signOutWithUser(composeTestRule)
+  }
+
+  private fun simulateResetPassword() {
```

Comment:
You might want to extract this method, this will avoid duplication, Or even simply combine the two tests files `ResetPasswordSettingsTest` and `ResetPasswordWelcomeTest` as they are technically very similar.

Reply:
I don't know how two e2e tests are handled in the same file but I can definitely extract the function.

Reply:
Just so you know, having two test classes in the same file produces the same behavior as having two separate files containing each class.
Also, if you chose extracting the method, consider simply placing it in the `EndToEndTest` class so that all other end to end tests can access it.

Reply:
Thank you for the clarification